scripts/check_catalog_urls.py

```
@@ -0,0 +1,150 @@
+
+def safe_url(url):
+    parts = urllib.parse.urlsplit(url)
+    return urllib.parse.urlunsplit(
+        parts._replace(path=urllib.parse.quote(parts.path, safe="/:@%"))
+    )
+
+
+def check_url(url, timeout=20, max_retries=4):
+    """Return (status_code, error_message). Code 0 means network failure."""
+    req = urllib.request.Request(safe_url(url), method="HEAD",
+                                 headers={"User-Agent": "AAVA-catalog-url-check/1.0"})
```

Ruff (0.15.11)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)